conf/config.go

```diff
@@ -0,0 +1,193 @@
+package conf
+
+/*
+   This is a package that wraps the viper, a package designed to handle config
+   files, for the BCDA app. This package will go through three different stages.
+
+   1. Local env looks primarily at conf package for variables, but will also look
+   in the environment for any variables it is not tracking. PROD/TEST/DEV will
+   only look in the environment. (WE ARE HERE NOW)
+
+   2. Local env will only look at conf package. PROD/TEST/DEV will look at both.
+
+   3. All env will look at the conf package.
+
+   Assumptions:
+   1. The configuration file is a env file (can be changed if needed).
+   2. The configuration variables, once it has been ingested by the conf package,
+   will stay immutable during the uptime of the application (exception is test).
+*/
+
+import (
+	"os"
+	"testing"
+
+	"github.com/spf13/viper"
+)
+
+// Private global variable:
+
+// An instance of the viper struct containing the config information. Only made
+// accessible through public functions: GetEnv, SetEnv, etc.
+var envVars viper.Viper
+
+// Implementing a state machine that tracks the status of the config ingestion.
+// This state machine should go away when in stage 3.
+const (
+	configgood    uint8 = 0
+	configbad     uint8 = 1
+	noconfigfound uint8 = 2
+)
+
+// if config file found and loaded, doesn't changed
+var state uint8 = configgood
+
+/*
+   This is the private helper function that sets up viper. This function is
+   called by the init() function only once during initialization of the package.
+*/
+func setup(dir string) *viper.Viper {
+
+	// Viper setup
+	var v = viper.New()
+	v.SetConfigName("local")
+	v.SetConfigType("env")
+	v.AddConfigPath(dir)
+	// Viper is lazy, do the read and parse of the config file
+	var err = v.ReadInConfig()
+
+	// If viper cannot read the configuration file...
+	if err != nil {
+		state = configbad
+	}
+
+	return v
+
+}
+
+/*
+   init:
+   First thing to run when this package is loaded by the binary.
+   Even if multiple packages import conf, this will be called and ran ONLY once.
+*/
+func init() {
+
+	// Possible config file locations: local and PROD/DEV/TEST respectfully.
+	var locationSlice = [2]string{
+		"/go/src/github.com/CMSgov/bcda-app/shared_files/decrypted",
+		"/go/src/github.com/CMSgov/DoesNotExistYet", // This is a placeholder for now
```

Comment:
Do we need to specify an extra path that will fail? Maybe if we have a comment instead to describe that we could possibly add new paths for the various environments that would be just as effective?